### PR TITLE
chore: Add color contrast recommendations end to end tests

### PR DESCRIPTION
**Pull request by sfoslund**
#### Details

Add end to end tests for color contrast recommendations. For this PR, just adding simple test that recommendation is displayed for a failing page. 

##### Motivation

Addresses issue #4379

##### Context

#4360 added color contrast recommendations but no end to end tests were added. This PR adds e2e tests for this feature.

#### Pull request checklist

- [x] Addresses an existing issue: #4379
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [n/a] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

**Comment by sfoslund** [comment]
@jalkire since you filed this original issue, can you please review? I made this change to familiarize myself with this repo, so please let me know if I'm not following any conventions or anything. Also, were there more in depth tests that this issue was tracking adding? Is there anything else from this feature that needs coverage?

File: src/tests/end-to-end/tests/target-page/color-contrast.test.ts
Comment (jalkire) [inline src/tests/end-to-end/tests/target-page/color-contrast.test.ts:59]: Another way you could do this is with a snapshot of the instanceCard, something like:
```suggestion
        const instanceCard = await detailsViewPage.getOuterHTMLOfSelector(
            fastPassAutomatedChecksSelectors.instanceCard,
        );
        expect(instanceCard).toMatchSnapshot();
```
This is a little more concise and makes updating the test easy if we tweak the text later. It's possible you'd want to add a new selector more specific to the color contrast result if you went this route. 

All that said, if you find the more explicit text validation preferable, that works just fine too 🙂

Comment (sfoslund) [reply src/tests/end-to-end/tests/target-page/color-contrast.test.ts]: Makes sense to me, the latest commit should fix both these issues.

**Review by jalkire — APPROVED** [review]
👍

**Comment by sfoslund** [comment]
@jalkire do you recognize the failure in web-e2e-reliability? It seems to be failing in other open PRs so I don't think it's related to this change. Also, what's the merge etiquette for this repo?

**Comment by jalkire** [comment]
> @jalkire do you recognize the failure in web-e2e-reliability? It seems to be failing in other open PRs so I don't think it's related to this change. Also, what's the merge etiquette for this repo?

For the e2e reliability, I agree that those errors don't look related to your change. I reran the failed tests on the build, but you should be good to go.

For merge etiquette, you're good to merge when ready once you've got the necessary approval! One thing to confirm when merging is that the final merge commit title starts with a semantic tag (as seen in the PR checklist). Some people paste the PR description into the extended description, others leave it as is--that's up to you

**Pull request merged**
